### DISCUSSION TITLE
ForwardedMessages and ReplyMessages (whatsapp).

### DISCUSSION THREAD

**Discussion by gisms** [sepinf-inc/IPED#1013] [General]
Hi everyone!

I was researching about forwardedMessages and ReplyMessages (whatsapp). 
I found bug #997 which more or less describes the issue that we are experiencing.

Version 18.13: Decoding via external parser, ufdr forwardedMessages and ReplyMessages are flagged with an attachment. Decoding via ufdr internal parser, duplicates it and does not flag.

I build a iped version from source today, but only the duplication problem resolved, there is still no indication that it is a forwarded or reply message. And the external parser stopped to flagged it with attachment (wich I dont think is a good way).

Is this the scenario for now?
Would it be worth opening an issue to request that these types of messages, when decoded via external at least, to be flagged in a specific way?

**Comment by lfcnassif** [comment]
Please test latest 3.18.x snapshot: https://github.com/sepinf-inc/IPED/actions/runs/1982951751 and check if it works.

> And the external parser stopped to flagged it with attachment (wich I dont think is a good way).

They are not attachments, past behavior was wrong.

> Is this the scenario for now?

Now there are new properties ufed:ForwardedMessageData and ufed:ReplyMessageData you can use to filter by those messages types e.g.: _ufed:ForwardedMessageData:*_ this will list forwarded instant messages in **table tab**.

> Would it be worth opening an issue to request that these types of messages, when decoded via external at least, to be flagged in a specific way?

We already have an open issue to render them properly in Preview tab: #998. This is not scheduled for a future 3.18.x release. Of course, contributions are always welcome.

**Reply by gisms** [reply]
In this version you sent and the one I built today, using external parser, it still duplicates for my case.

About the attachment, I expressed myself badly, sorry. I agree that the attachment is not the correct type of signaling, but at the same time I think that the absence of a type of signaling, in the html chat, can generate questioning and misinterpretations. I understand that there is a priority queue and few hands.

**Reply by gisms** [reply]
With any type of sign, in the html chat, at least I can explain in a external report.

**Reply by lfcnassif** [reply]
Release process for 3.18.14 already started, so, it will have to wait...

**Reply by lfcnassif** [reply]
I don't think it is a bug to stop it... just lack of functionality

**Reply by gisms** [reply]
> I don't think it is a bug to stop it... just lack of functionality

Yes, true.

**Reply by lfcnassif** [reply]
> In this version you sent and the one I built today, using external parser, it still duplicates for my case.

I can't reproduce this with latest 3.18.x link I sent and some ufdr samples I have here. Could you share your ufdr file privately, so I can confirm the issue and test a possible fix?

**Reply by lfcnassif** [reply]
Maybe have you used phoneParsersToUse = all? If that's the case, chats being duplicated is expected behavior. If not, I really need some sample ufdr to reproduce the issue and try to fix it.

**Reply by gisms** [reply]
I will do more tests with different versions and also models of cell phones and return. it will take longer but I'll be back
